turbopack/crates/turbopack-ecmascript/src/transform/mod.rs

```diff
@@ -219,33 +219,16 @@ impl EcmascriptInputTransform {
                     },
                 );
 
-                let module_program = std::mem::replace(program, Program::Module(Module::dummy()));
-
-                let module_program = if let Program::Script(Script {
-                    span,
-                    mut body,
-                    shebang,
-                }) = module_program
-                {
-                    Program::Module(Module {
-                        span,
-                        body: body.drain(..).map(ModuleItem::Stmt).collect(),
-                        shebang,
-                    })
-                } else {
-                    module_program
-                };
-
                 // Explicit type annotation to ensure that we don't duplicate transforms in the
                 // final binary
-                *program = module_program.apply((
+                program.mutate((
                     preset_env::transform_from_env::<&'_ dyn Comments>(
-                        top_level_mark,
+                        unresolved_mark,
                         Some(&comments),
                         config,
                         Assumptions::default(),
                     ),
-                    inject_helpers(unresolved_mark),
+                    inject_helpers(top_level_mark),
```

Comment:
@kdy1 seems like these were the wrong way around?
Didn't seem to cause any issue so far, but probably still wrong?
![Bildschirmfoto 2025-06-20 um 19 42 24](https://github.com/user-attachments/assets/b0c45020-d9e1-4287-b6c4-09e95b6fa62b)

Reply:
Oops. cc @magic-akari

Reply:
I'll fix it. It was turbopack issue, not SWC core issue. Sorry for the ping.

Reply:
What do you mean? Isn't this already the fix?